src/main/java/com/thealgorithms/datastructures/graphs/WelshPowell.java

```diff
@@ -0,0 +1,73 @@
+package com.thealgorithms.datastructures.graphs;
+
+import java.util.LinkedList;
+
+/*
+ *  The Welsh-Powell algorithm is a graph coloring algorithm
+ *  used for coloring a graph with the minimum number of colors.
+ *  https://en.wikipedia.org/wiki/Graph_coloring
+ */
+public class WelshPowell {
+
+    public static class WPGraph {
+
+        private int numVer; // Number of vertices in the graph
+        public LinkedList<Integer>[] adjLists; // Adjacency lists for the graph
+
+        // Constructor to initialize the graph with a certain number of vertices
+        public WPGraph(int vertices) {
+            numVer = vertices;
+            adjLists = new LinkedList[vertices];
+
+            for (int i = 0; i < vertices; i++) {
+                adjLists[i] = new LinkedList<>();
+            }
+        }
+
+        // Method to add an edge to the graph (undirected)
+        public void addEdge(int src, int dest) {
+            // Check to prevent duplicate edges
+            if (!adjLists[src].contains(dest)) {
+                adjLists[src].add(dest);
+                adjLists[dest].add(src); // For undirected graphs, add the reverse edge as well
+            }
+        }
+
+        // Method implementing Welsh Powell coloring algorithm
+        public int[] welshPowellColoring() {
```

Comment:
What do you think about having `welshPowellColoring` as a method of `WelshPowell` taking an object of `WPGraph` as an input?

Reply:
I am currently trying to implement it on my code but i run into some issues regarding the test class and how it can access wpgraph and i am trying to fix it.  If it is not necessary let me know so i just implement the nodeA,nodeB change and the Hashset instead of the Linked list